File: Novels/packages/5-mobile-ios/StoryReader/Model/Annotation.swift
import Foundation

// Annotation model - Text highlights + notes
struct Annotation: Codable, Identifiable {
    let id: String
    let userId: String
    let storyId: String
    let chapterId: String
    let paragraphIndex: Int
    let selectedText: String
    let note: String?
    let color: String? // Hex color
    let tags: [String]?
    let createdAt: Date
    let updatedAt: Date?
    let syncedAt: Date?
    let syncStatus: SyncStatus
    
    enum SyncStatus: String, Codable {
        case synced
        case pending
        case conflict
        case error
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case storyId = "story_id"
        case chapterId = "chapter_id"
        case paragraphIndex = "paragraph_index"
        case selectedText = "selected_text"
        case note, color, tags
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case syncedAt = "synced_at"
        case syncStatus = "sync_status"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Annotation.kt
package com.storysphere.storyreader.model

data class Annotation(
    val id: String,
    val userId: String,
    val storyId: String,
    val chapterId: String,
    val startPosition: Int,
    val endPosition: Int,
    val selectedText: String,
    val note: String? = null,
    val color: String? = null,
    val tags: List<String> = emptyList(),
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

